qualimap/qualimap.cwl

```diff
@@ -0,0 +1,104 @@
+#!/usr/bin/env cwl-runner
+cwlVersion: v1.0
+class: CommandLineTool
+
+# Metadata
+label: qualimap-qc
+doc: This is qualimap CWL tool definition http://qualimap.bioinfo.cipf.es/
+
+hints:
+  ResourceRequirement:
+    ramMin: 4000
+    coresMin: 1
+  SoftwareRequirement:
+    packages:
+      qualimap:
+        version: [ "2.2.2d" ]
+        specs:
+          - https://identifiers.org/rrid/RRID:SCR_001209
+          - https://bio.tools/qualimap
+  DockerRequirement:
+    dockerPull: 'quay.io/biocontainers/qualimap:2.2.2d--1'
+requirements:
+  - class: ShellCommandRequirement
+
+# Base command
+baseCommand: [ "qualimap"]
+
+# Perform RNA-seq QC analysis on paired-end data http://qualimap.bioinfo.cipf.es/doc_html/command_line.html
+# Desired java memory size can be set using --java-mem-size option
+arguments:
+  - valueFrom: |-
+      rnaseq --paired --java-mem-size="$(inputs.javamem)"
+    shellQuote: false
+
+inputs:
+  javamem:
+    type: string
+    doc: |
+      Set desired Java heap memory size
+
+  outdir:
+    type: string
+    inputBinding:
+      prefix: "-outdir"
+    doc: |
+      Output folder for HTML report and raw data.
+
+  algo:
+    type: string
+    inputBinding:
+      prefix: "-a"
+    doc: |
+      Counting algorithm:
+      uniquely-mapped-reads(default) or proportional.
+
+  inputBam:
+    type: File
+    secondaryFiles: .bai
+    inputBinding:
+      prefix: "-bam"
+    doc: |
+      Input mapping file in BAM format.
+
+  seqProtocol:
+    type: string
+    inputBinding:
+      prefix: "-p"
+    doc: |
+      Sequencing library protocol:
+      strand-specific-forward, strand-specific-reverse or 
+      non-strand-specific (default).
+  
+  gtf:
+    type: File
+    inputBinding:
+      prefix: "-gtf"
+    doc: |
+      Region file in GTF, GFF or BED format. 
+      If GTF format is provided, counting is based on
+      attributes, otherwise based on feature name.
+
+outputs:
+  qualimapQC:
+    type: Directory
+    outputBinding:
+      glob: "$(inputs.outdir)" 
```

Comment:
```suggestion
  qualimapQC:
    type: Directory
    label: HTML report and raw data
    outputBinding:
      glob: $(inputs.inputBam.nameroot)
```

To match my other suggestion

Are there any specific files in this "HTML report and raw data" folder that you might want to have as individual outputs as well?

The `qualimap rnaseq --help` output suggests that a PDF report is also written to `report.pdf` (the default value for `-outfile`). If that is still produced when `-outdir` is used, then lets also make that available to users.

Reply:
I didn't find a `report.pdf`. HTML report is something that is mostly looked at first. I'll push a commit to explicitly capture it.

Reply:
Added in [24da101](https://github.com/common-workflow-library/bio-cwl-tools/pull/78/commits/24da101d997fa6e26a7a21dc62de11eee89c1d06)